File: VB/04_Shapes/UngroupShapes.vb
Imports System.Collections
Imports Spire.Presentation
Imports Spire.Presentation.Drawing

Namespace UngroupShapes
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create a PPT document
			Dim ppt As New Presentation()
			'Load the document from disk
			ppt.LoadFromFile("..\..\..\..\..\..\Data\GroupShapes.pptx")
			'Get the GroupShape
			Dim groupShape As GroupShape = TryCast(ppt.Slides(0).Shapes(0), GroupShape)
			'Ungroup the shapes
			ppt.Slides(0).Ungroup(groupShape)
			'Save the document
			Dim result As String = "UngroupShapes.pptx"
			ppt.SaveToFile(result, Spire.Presentation.FileFormat.Pptx2013)
			PresentationDocViewer(result)
		End Sub

		Private Sub PresentationDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/04_Shapes/UngroupShapes.cs
using System;
using System.Collections;
using System.Drawing;
using System.Windows.Forms;
using Spire.Presentation;
using Spire.Presentation.Drawing;

namespace UngroupShapes
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a PPT document
            Presentation ppt = new Presentation();
            //Load the document from disk
            ppt.LoadFromFile(@"..\..\..\..\..\..\Data\GroupShapes.pptx");
            //Get the GroupShape
            GroupShape groupShape = ppt.Slides[0].Shapes[0] as GroupShape;
            //Ungroup the shapes
            ppt.Slides[0].Ungroup(groupShape);
            //Save the document
            String result = "UngroupShapes.pptx";
            ppt.SaveToFile(result, Spire.Presentation.FileFormat.Pptx2013);
            PresentationDocViewer(result);
		   }
	
		private void PresentationDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}